cg/cli/workflow/microsalt/base.py

```diff
@@ -10,149 +10,117 @@
+@click.option("-d", "--dry-run", is_flag=True, help="print to console")
+@click.option("-t", "--ticket", help="include all microbial samples for a ticket")
 @click.pass_context
-def microsalt(context: click.Context, order_id):
+def microsalt(context: click.Context, ticket, dry_run):
     """Microbial workflow"""
     context.obj["db"] = Store(context.obj["database"])
     hk_api = hk.HousekeeperAPI(context.obj)
     lims_api = lims.LimsAPI(context.obj)
-    deliver = DeliverAPI(
-        context.obj,
-        hk_api=hk_api,
-        lims_api=lims_api,
-        case_tags=PROJECT_TAGS,
-        sample_tags=SAMPLE_TAGS,
-    )
     context.obj["api"] = AnalysisAPI(db=context.obj["db"], hk_api=hk_api, lims_api=lims_api)
     context.obj["lims_microsalt_api"] = LimsMicrosaltAPI(lims=lims_api)
 
     if context.invoked_subcommand is None:
-        if order_id is None:
-            LOG.error("Please provide an order")
+        if not ticket:
+            LOG.error("Please provide a ticket")
             context.abort()
         else:
             # execute the analysis!
-            context.invoke(config_case, order_id=order_id)
-            context.invoke(link, order_id=order_id)
-            context.invoke(run, order_id=order_id)
+            context.invoke(config_case, ticket=ticket, dry_run=dry_run)
+            context.invoke(link, ticket=ticket, dry_run=dry_run)
+            context.invoke(run, ticket=ticket, dry_run=dry_run)
 
 
 @microsalt.command()
-@click.option("-o", "--order", "order_id", help="link all microbial samples for an order")
+@click.option("-d", "--dry-run", is_flag=True, help="only print to console")
+@click.option("-t", "--ticket", help="link all microbial samples for a ticket")
 @click.argument("sample_id", required=False)
 @click.pass_context
-def link(context: click.Context, order_id: str, sample_id: str):
+def link(context: click.Context, dry_run: bool, ticket: str, sample_id: str):
     """Link microbial FASTQ files for a SAMPLE_ID"""
-    sample_objs = None
-
-    if order_id and (sample_id is None):
-        # link all samples in a case
-        sample_objs = context.obj["db"].microbial_order(order_id).microbial_samples
-    elif sample_id and (order_id is None):
-        # link sample in all its families
-        sample_objs = [context.obj["db"].microbial_sample(sample_id)]
-    elif sample_id and order_id:
-        # link only one sample in a case
-        sample_objs = [context.obj["db"].microbial_sample(sample_id)]
+
+    api = context.obj["api"]
+    sample_objs = api.get_samples(ticket, sample_id)
 
     if not sample_objs:
-        LOG.error("provide order and/or sample")
+        LOG.error("provide ticket and/or sample")
         context.abort()
 
     for sample_obj in sample_objs:
         LOG.info("%s: link FASTQ files", sample_obj.internal_id)
-        context.obj["api"].link_sample(
+        if dry_run:
+            continue
+        api.link_sample(
```

Comment:
Here for example a FastqHandler is instantiated for every sample in the loop. Suggest to do:
```python
def link_microbial_samples(samples: List[Sample], ticket: str, fastq_handler: FastqHandler, api: SomeAPI):
    """Links a list of microbial samples"""
    ...
```

Reply:
What did you think of this @patrikgrenfeldt ?